scipy/sparse/tests/test_base.py

```diff
@@ -59,6 +59,23 @@ def assert_array_equal_dtype(x, y, **kwargs):
     assert_(x.dtype == y.dtype)
     assert_array_equal(x, y, **kwargs)
 
+def _underlying_arrays(x):
+    arrays = []
+    for a in x.__dict__.values():
+        if isinstance(a, (np.ndarray, np.generic)):
+            arrays.append(a)
+    return arrays
+
+def format_can_share_memory(format):
```

Comment:
I'd prefer this as a frozenset constant, rather than a function. Maybe `ARRAY_BACKED_FORMATS`?